src/main/java/org/generationcp/middleware/dao/dms/ProgramFavoriteDAO.java

```diff
@@ -88,6 +90,22 @@
 		}
 	}
 
+	public List<ProgramFavorite> getProgramFavorites(final String programUUID, final ProgramFavorite.FavoriteType favoriteType, final Set<Integer> entityId)
```

Comment:
1. rename to entityIds
2. I think we should allow null entityIds to get all favorites by program and type. Not sure if this hibernate version properly handles null in a Restrictions.in. Could you please check?